src/OpenAI.jl

```diff
@@ -233,6 +233,57 @@ function create_completion(api_key::String, model_id::String; http_kwargs::Named
     return openai_request("completions", api_key; method="POST", http_kwargs=http_kwargs, model=model_id, kwargs...)
 end
 
+
+"""
+Checking for JSON validity of the response body used in the stream
+"""
+isvalidjson(str) =
+    try
+        JSON3.read(str)
+        true
+    catch
+        false
+    end
```

Comment:
Minor note but I typically prefer `function ... end` syntax for multiline functions.

```suggestion
function isvalidjson(str)
    try
        JSON3.read(str)
        true
    catch
        false
    end
end
```